infrastructure/terragrunt.hcl

```diff
@@ -60,6 +74,8 @@ terraform {
 
   required_providers {
     ${local.required_provider_azure}
+    ${local.required_provider_azuread}
```

Comment:
Are these required by all modules? Otherwise you could move this just into the specific one's terragrunt.hcl file that needs them

Reply:
Looking at your code you already do this as well, so you can just remove it as a required provider here

Reply:
yep, tidied these up and moved dbks to a var too.